### ISSUE TITLE
Datadocs on S3 is always overwritten when GE runs from GitHub Action

### ISSUE THREAD

**Issue by tomas-sobotik** [fivetran/great_expectations#2581]
**Describe the bug**
I have my DataDocs stored in S3 bucket. Everything works as expected when i run the checkpoint from my local machine. I can see in the data docs the complete history of all runs. But when I run the checkpoint as part of my GitHub action it always overwrite the content of data_docs subdirectory in S3 and store there just the latest run, which means that all history is gone. I have enabled server access logging for S3 bucket and try to dig into the logs to find out what is happening behind the scenes and it looks like when the checkpoint validation runs through github action there are sent to S3 only api request to PUT and DELETE objects.

`REST.PUT.OBJECT key ...`
`BATCH.DELETE.OBJECT key ..`

It does not check the versioning, etc. Just simply remove everything and store there the latest run.
I was trying the same from my local machine and this time it works as expected. It keeps older data docs files in S3 and just adding the new ones. When I was checking the S3 server access logging. This time it does not do any DELETE but a lot of “verification” around is done before new objects are inserted:

`REST.GET.LOCATION`
`REST.GET.BUCKET`
`REST.GET.VERSIONING`
`REST.HEAD.BUCKET`
`REST.GET.OBJECT_LOCK_CONFIGURATION`
`REST.PUT.OBJECT`
`REST.PUT.OBJECT`
`...`

I do not use the GE github action because that one use GE version 0.11.5 but i use GE 0.12.1 so I have developed my own github action which does following:

- install python
- install great expectations
- install sql alchemy
- install sql alchemy for snowflake (my DB)
- my repository checkout
- running the checkpoint 

I can provide action yaml file.

**To Reproduce**
1. Define datadocs storage on S3 bucket
2. Run checkpoint through github action multiple times
3. Verify in s3 if you have complete history of runs or just the latest run

**Expected behavior**
I would expect that data_docs will keep history and just will be adding new checkpoint runs.

**Environment (please complete the following information):**
 - Operating System: GitHub action is running on Ubuntu-latest, my local is Mac OS 10.15.7
 - Great Expectations Version: 0.12.1
 - Python version: 3.9.1

**Additional info:**
Trying different ways how to bypass this. One of my attempt was store data_docs in git repository and do the checkout into the github runner as part of the github action. I have been checking content of data_docs inside the GH action and i could see my history is populated from git repository but when i run `great_expectations checkpoint run` which is supposed to update the data_docs, it again replaces the data_docs content on github runner and keep there only the latest run.

I have been also trying the same with `aws s3 sync` to first download the history from s3 into the github runner, run the checkpoint and again do manually `aws s3 sync` to upload the latest run back into the s3. This also not work fully. I can see the new and also old runs under data_docs/validations but index.html still contains only the latest run.

Another attempt. I've tried to store also validations in git repository and maybe now it looks how i would expect. Newly build data_docs contains history of all my runs + new actual run. So maybe also validations needs to be available in time of data_docs generation. I will investigate it more.

**Comment by tomas-sobotik** [comment]
I have found the root cause. It behaves this way when you have data docs stored in S3 and validations stored locally and run it via GitHub action. When you have this way, you have only latest validations stored in github runner running your git hub action that's why you have only latest run available in data docs. I have found two ways how to solve it.

1. Store also validations on s3
2. Store validations in git repository - you do git checkout on github runner anyway so when you have validations on git you will make them available on github runner.
